File: ios/Framework/Core/Auth/SignUpScreenlet/SignUpViewModel.swift
/**
* Copyright (c) 2000-present Liferay, Inc. All rights reserved.
*
* This library is free software; you can redistribute it and/or modify it under
* the terms of the GNU Lesser General Public License as published by the Free
* Software Foundation; either version 2.1 of the License, or (at your option)
* any later version.
*
* This library is distributed in the hope that it will be useful, but WITHOUT
* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
* FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
* details.
*/
import UIKit


@objc public protocol SignUpViewModel {

	var emailAddress: String? { get set }
	var screenName: String? { get set }
	var password: String? { get set }
	var firstName: String? { get set }
	var middleName: String? { get set }
	var lastName: String? { get set }
	var jobTitle: String? { get set }

	var editCurrentUser: Bool { get set }

}

File: xamarin/Framework/LiferayScreens.iOS/Auth/SignUp/SignUpViewModel.cs
using Foundation;
using ObjCRuntime;
using System;

namespace LiferayScreens
{
    interface ISignUpViewModel {}

    // @protocol SignUpViewModel
    [BaseType(typeof(NSObject))]
    [Protocol, Model]
    interface SignUpViewModel
    {
        // @required @property (copy, nonatomic) NSString * _Nullable emailAddress;
        [Abstract]
        [NullAllowed, Export("emailAddress")]
        string EmailAddress { get; set; }

        // @required @property (copy, nonatomic) NSString * _Nullable screenName;
        [Abstract]
        [NullAllowed, Export("screenName")]
        string ScreenName { get; set; }

        // @required @property (copy, nonatomic) NSString * _Nullable password;
        [Abstract]
        [NullAllowed, Export("password")]
        string Password { get; set; }

        // @required @property (copy, nonatomic) NSString * _Nullable firstName;
        [Abstract]
        [NullAllowed, Export("firstName")]
        string FirstName { get; set; }

        // @required @property (copy, nonatomic) NSString * _Nullable middleName;
        [Abstract]
        [NullAllowed, Export("middleName")]
        string MiddleName { get; set; }

        // @required @property (copy, nonatomic) NSString * _Nullable lastName;
        [Abstract]
        [NullAllowed, Export("lastName")]
        string LastName { get; set; }

        // @required @property (copy, nonatomic) NSString * _Nullable jobTitle;
        [Abstract]
        [NullAllowed, Export("jobTitle")]
        string JobTitle { get; set; }

        // @required @property (nonatomic) BOOL editCurrentUser;
        [Abstract]
        [Export("editCurrentUser")]
        bool EditCurrentUser { get; set; }
    }
}
